internal/resources/connections/metrics_endpoint_scrape_job_test.go

```diff
@@ -130,3 +142,39 @@ resource "grafana_connections_metrics_endpoint_scrape_job" "test" {
   scrape_interval_seconds       = 120
 }
 `
+
+var resourceWithForEachValidURL = `
+locals {
+  jobs = [
+    { name = "test", url = "https://google.com" }
+  ]
+}
+
+resource "grafana_connections_metrics_endpoint_scrape_job" "valid_url" {
+  for_each = { for j in local.jobs : j.name => j.url }
+  stack_id = "......"
+  name = each.key
+  enabled = false
+  authentication_method = "bearer"
+  authentication_bearer_token = "test"
+  url = each.value
+}
+`
+
+var resourceWithForEachInvalidURL = `
```

Comment:
So with the [early break](https://github.com/grafana/terraform-provider-grafana/pull/1942/files#diff-9074a61dbecc623cb6d2571cc8efdf409bd20e2f72af09153891b16b665fa0a3R35-R37) on `IsNull` or `IsUnknown`, we still do the validation at a later point?

Reply:
Yes, this example provides an empty string so it continues to https://github.com/grafana/terraform-provider-grafana/blob/340c6908a98ced498eee023261220a682003d471/internal/resources/connections/resources.go#L42

(unit test on empty string: https://github.com/grafana/terraform-provider-grafana/blob/e5e3eb356e1d2e4c99b3d862dee1889ffd5ca9af/internal/resources/connections/resources_test.go#L34-L33)